### PR TITLE
test: Add domain tests for wholesale events

### PR DESCRIPTION
**Pull request by dstenroejl**
- [x] Test one AmountPerChargeResultProducedV1 event contains expected timeseries point values
- [x] Test one MonthlyAmountPerChargeResultProducedV1 event contains expected amount

Part of:
https://app.zenhub.com/workspaces/mandalorian-5fd22a1a59e4740018f5ab5a/issues/gh/energinet-datahub/green-energy-hub/444

### CODE REVIEW

File: source/dotnet/domain-tests/Features/Calculations/TestData/amount_for_es_for_hourly_tarif_40000_for_e17_e02.csv [hunk 1]
```diff
@@ -0,0 +1,601 @@
+grid_area;energy_supplier_id;quantity;time;price;amount;charge_code;
```
Comment (BjarkeMeier) [inline source/dotnet/domain-tests/Features/Calculations/TestData/amount_for_es_for_hourly_tarif_40000_for_e17_e02.csv:1]: How do we know that these values are correct?
What happens when we move to dev-001?

File: source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs [hunk 2]
```diff
@@ -225,5 +226,65 @@ public void AndThen_ReceivedMonthlyAmountPerChargeResultProducedEventsCountIsEqu
             // Assert
             Fixture.ScenarioState.ReceivedMonthlyAmountPerChargeResultProducedV1.Count.Should().Be(expected);
         }
+
+        /// <summary>
+        /// Notice we don't verify 'TimeSeriesPoints.QuantityQualities' in this scenario.
```
Comment (BjarkeMeier) [inline source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs:231]: Why?
Add the reason in the code doc?

Comment (BjarkeMeier) [reply source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs:231]: Found it in the implementation comments. That's probably fine.

File: source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs [hunk 3]
```diff
@@ -225,5 +226,65 @@ public void AndThen_ReceivedMonthlyAmountPerChargeResultProducedEventsCountIsEqu
...
+                && item.ChargeOwnerId == expectedChargeOwnerId
+                && item.SettlementMethod == expectedSettlementMethod);
+
+            using var assertionScope = new AssertionScope();
+            actualEvents.Should().HaveCount(1);
+
+            var actualEvent = actualEvents.First();
+            actualEvent.TimeSeriesPoints.Should().HaveCount(expectedTimeSeriesPoints.Count);
+
+            // We clear incomming 'QuantityQualities' before comparing with test data, because we don't have them in our test data file.
```
Comment (BjarkeMeier) [inline source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs:258]: Should we add the qualities?

File: source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs [hunk 4]
```diff
@@ -225,5 +226,65 @@ public void AndThen_ReceivedMonthlyAmountPerChargeResultProducedEventsCountIsEqu
...
+                    return item;
+                })
+                .Should().BeEquivalentTo(expectedTimeSeriesPoints);
+        }
+
+        [ScenarioStep(12)]
+        [DomainFact]
+        public void AndThen_OneSpecificMonthlyAmountPerChargeResultProducedEventContainsExpectedMonthlyAmount()
+        {
+            var expectedEnergySupplierId = "5790001687137";
```
Comment (BjarkeMeier) [inline source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs:272]: Observation: We have u-001 data scattered over all our domain test code.
Should we segregate the data from the tests?

File: source/dotnet/domain-tests/Features/Calculations/TestData/amount_for_es_for_hourly_tarif_40000_for_e17_e02.csv (hunk 1 above)
Comment (dstenroejl) [reply source/dotnet/domain-tests/Features/Calculations/TestData/amount_for_es_for_hourly_tarif_40000_for_e17_e02.csv:1]: Ad. 1) Our SME has given us the data.
Ad. 2) We will have to update several things when moving to dev-001, updating test data is one of them. This is documented in Confluence which is not public, so I will not link to it here.

File: source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs (hunk 3 above)
Comment (dstenroejl) [reply source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs:258]: Not at the moment.

File: source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs (hunk 4 above)
Comment (dstenroejl) [reply source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs:272]: Not a the moment. But its something that is relevant to take into consideration when we move into dev-001, and its also on the QA guilds radar.

File: source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs (hunk 2 above)
Comment (dstenroejl) [reply source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs:231]: It was a scoping decision.

Comment (BjarkeMeier) [reply source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs:231]: Out of scope for PR or story? Do we have plans on adding it later?

Comment (dstenroejl) [reply source/dotnet/domain-tests/Features/Calculations/WholesaleFixingCalculationScenario.cs:231]: Out-of-cope for PR and story. It was added on another story. This is a public repository, so I won't link to the story here.

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**